packages/components/src/components/hds/button-set/index.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import templateOnlyComponent from '@ember/component/template-only';
+
+const HdsButtonSetComponent = templateOnlyComponent();
```

Comment:
(feel free to take or leave) If we wanted to express to a consumer that this component takes yielded content, we could add something like
```
interface HdsButtonSetSignature {
  Blocks: { default: [] }
}
const HdsButtonSetComponent = templateOnlyComponent<HdsButtonSetSignature>()
```

Reply:
@alex-ju Should I add this?

Reply:
@natmegs I have the same doubt as @KristinLBradley because if we add a signature TS complains that `templateOnlyComponent` doesn't accept type arguments (`Expected 0 type arguments, but got 1`)
<img width="658" alt="modified" src="https://github.com/hashicorp/design-system/assets/686239/0ad12c22-1097-4683-ab70-410f0a3a5968">
so I am not sure the usefulness of a `template-only` declaration: in which _real_ use case would be used?

Reply:
Oh good to know! I was just going off the docs [here](https://typed-ember.gitbook.io/glint/environments/ember/template-only-components). If the types don't allow passing the component signature then IMO it's not super useful.

An alternative would be:
```
interface HdsButtonSetSignature {
  Blocks: { default: [] }
}
export default class HdsButtonSetComponent extends Component<HdsButtonSetSignature> {}
```

Reply:
Yes, that's what I started to do too 👍

Reply:
@aklkv do you know if the documentation linked by @natmegs is wrong? should we open an issue somewhere?

Reply:
this is a known issue – if the repo contains a dependency on [@types/ember__component@v3](https://www.npmjs.com/package/@types/ember__component/v/3.16.6?activeTab=code) (and unfortunately we do, at the workspace level, via a third party dependency – can't remember which one, but I can look again). if you look under the FAQ here, [the second question](https://github.com/ijlee2/ember-codemod-args-to-signature/?tab=readme-ov-file#frequently-asked-questions) you'll see they suggest the same options you considered. In the existing code we took the approach @natmegs and @didoo mentioned.

I tried [resolving it](https://github.com/hashicorp/design-system/pull/2023/commits/9abb66b892b431f22fe932e0f391fd87dc1c77e0) a while back, but it didn't stick for some reason.

Reply:
Should we document it in https://github.com/hashicorp/design-system/pull/2087 then?

Reply:
@alex-ju also, would it work if we forced the version of `@types/ember__component` in the `resolutions` field of `package.json`?

Reply:
Applied a fix to this – until we manage to get rid of `@types/ember__component@v3` from our dependencies.  @KristinLBradley after #2113 gets merged you can follow the docs/codemod/Natalie's suggestion on this

Reply:
Thanks for the feedback everyone. I added a Signature following @natmegs' example above.

Reply:
Updated